### PR TITLE
Rotate implementation in Rust

### PR DESCRIPTION
**Pull request by surma**
Alright, here we go! Rotate implementation in Rust!

(FYI @ashleygwilliams)

### CODE REVIEW

File: src/codecs/rotate/processor.ts [hunk 1]
```diff
@@ -1,73 +1,26 @@
...
-  const out = new ImageData(outputWidth, outputHeight);
-  let i = 0;
+export async function rotate(
+  data: ImageData,
+  opts: RotateOptions,
+): Promise<ImageData> {
+  const flipDimensions = opts.rotate % 180 !== 0;
+  // Number of wasm memory pages (á 64KiB) needed to store the image twice.
+  const bytesPerImage = data.width * data.height * 4;
+  const numPagesNeeded = Math.ceil(bytesPerImage * 2 / (64 * 1024));
```
Comment (jakearchibald) [inline src/codecs/rotate/processor.ts]: We start at an offset of 4. Does this take that into account?

File: src/codecs/rotate/processor.ts [hunk 2]
```diff
@@ -1,73 +1,26 @@
...
+  const { instance } = (await (WebAssembly as any).instantiateStreaming(
+    fetch(wasmUrl),
+  )) as { instance: RotateModuleInstance };
```
Comment (jakearchibald) [inline src/codecs/rotate/processor.ts]: Our other wasm modules can be used many times per instantiation. Do we want to do the same here?

**Comment by surma** [comment]
There was a new Ubuntu and a new Rust image. I assume that’s where the difference came from. Making a new, up-to-date build.

**Comment by surma** [comment]
🎉

**Pull request merged**